implementations/utils/gpu.py

```diff
@@ -0,0 +1,113 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Utilities to create, delete, and SSH with GPUs."""
+
+import io
+from typing import Iterable, Tuple
+import uuid
+
+from absl import logging
+import airflow
+from airflow.decorators import task, task_group
+from airflow.utils.task_group import TaskGroup
+from apis import gcp_config, test_config
+import fabric
+import google.api_core.exceptions
+import google.auth
+# There is no GPU api to reserve for resources. So we use gcloud cmdline to reserve resouces.
+# import google.cloud.tpu_v2alpha1 as tpu_api
+import google.longrunning.operations_pb2 as operations
+from implementations.utils import ssh
+from implementations.utils.vm_api import gpu_api
+import paramiko
+
+
+@task
+def generate_gpu_name(base_gpu_name: str) -> str:
+  # note: GPU vm name need to match regex '(?:[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?)', while TPU vm allows '_'.
+  # return f'{base_gpu_name}-{str(uuid.uuid4())}'.replace('_', '-')
+  return f'gpu-{str(uuid.uuid4())}'
```

Comment:
I see the name in the comment is `return f'{base_gpu_name}-{str(uuid.uuid4())}'.replace('_', '-')`. Do we plan to use that or `f'gpu-{str(uuid.uuid4())}'`? I prefer the former one if possible.

Reply:
Thanks, I tried the former one before. However due to the `base_gpu_name` is too long, the full string will be long than expected (there is a requirement that some potion of the string should be less than 61 as in the regex mentioned).  Any ideas for a better naming?

Reply:
I see. If that's the case, could you add a comment indicates there is a limitation for len of characters?

Reply:
Comment added.